src/OrleansServiceBus/Providers/Streams/EventHub/TimePurgePredicate.cs

```diff
@@ -0,0 +1,47 @@
+﻿
+using System;
+
+namespace Orleans.ServiceBus.Providers
+{
+    /// <summary>
+    /// Determines if data should be purged based of time.
+    /// </summary>
+    public class TimePurgePredicate
+    {
+        private readonly TimeSpan minTimeInCache;
+        private readonly TimeSpan maxRelitiveMessageAge;
+
+        /// <summary>
+        /// Default time purge predicate never purges by time.
+        /// </summary>
+        public static readonly TimePurgePredicate Default = new TimePurgePredicate(TimeSpan.MinValue, TimeSpan.MaxValue);
+
+        /// <summary>
+        /// Contructor
+        /// </summary>
+        /// <param name="minTimeInCache">minimim age of data to keep in cache</param>
+        /// <param name="maxRelitiveMessageAge">maximum age of data to keep in the cache</param>
+        public TimePurgePredicate(TimeSpan minTimeInCache, TimeSpan maxRelitiveMessageAge)
+        {
+            this.minTimeInCache = minTimeInCache;
+            this.maxRelitiveMessageAge = maxRelitiveMessageAge;
+        }
+
+        /// <summary>
+        /// Checks to see if the message should be purged.
+        /// Message should be purged if it has been in the queue longer than the minTimeInCache, but it's relitive age is greater than maxRelitiveMessageAge.
+        /// </summary>
+        /// <param name="messageEnqueueUtc">time message was read from the queue</param>
+        /// <param name="cacheMaxEnqueueUtc">enque time of newest item in cache</param>
+        /// <param name="messageDequeueUtc">time message was enqueued</param>
```

Comment:
Is there are typo here? "messageDequeueUtctime message was enqueued, so is it enqueued or dequeued?